### ISSUE TITLE
cubic-to-spline: center of mass seems a bit "off"

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#514]
data: sct_testing_data/t2/t2_seg_labeled

result:
![screen shot 2015-09-21 at 10 59 50](https://cloud.githubusercontent.com/assets/2482071/9999415/51ceb8b2-6064-11e5-936a-3fde9bb430d5.png)

maybe the issue is related to the definition of the voxel center?

**Comment by benjamindeleener** [comment]
The command that produced the result was:
```
sct_label_utils -i t2_seg_labeled.nii.gz -t cubic-to-point -o t2_labeled_test.nii.gz
```

**Comment by benjamindeleener** [comment]
I am working on branch bdl_issue_514

**Comment by benjamindeleener** [comment]
I've refactored the whole code. It is much simpler now and fully use the capabilities of Coordinates objects (contained in msct_types).

See the result here:
![image](https://cloud.githubusercontent.com/assets/7757117/10022180/7e05fdd6-6119-11e5-94ad-2d5da7fc9895.png)

There is still an approximation during the process as the computed center of mass are float while indexes are integers:
````
Value = 2.0 : (30.0948275862, 52.9784482759, 26.0301724138) --> ( 30.0, 53.0, 26.0)
Value = 3.0 : (29.9421885754, 42.0110116999, 25.4218857536) --> ( 30.0, 42.0, 25.0)
Value = 4.0 : (30.2285251216, 24.6077795786, 24.9886547812) --> ( 30.0, 25.0, 25.0)
Value = 5.0 : (31.393812709, 8.27341137124, 24.8570234114) --> ( 31.0, 8.0, 25.0)
````

I've added a verbose option in order to display the text above and be sure that the approximation is not so bad.

**Comment by benjamindeleener** [comment]
I've added a test for center of mass computation:
![image](https://cloud.githubusercontent.com/assets/7757117/10023477/5ab45f9c-6120-11e5-8570-19781fb64d12.png)

**Comment by benjamindeleener** [comment]
Fixed in Pull Request #517
